extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ChunkInputStreamGenerator.java

```diff
@@ -144,6 +148,26 @@ static <T> ChunkInputStreamGenerator makeInputStreamGenerator(
                 if (type == Short.class) {
                     return ShortChunkInputStreamGenerator.convertBoxed(chunk.asObjectChunk(), rowOffset);
                 }
+                if (type == LocalDate.class) {
+                    return IntChunkInputStreamGenerator.<LocalDate>convertWithTransform(chunk.asObjectChunk(),
+                            rowOffset, date -> {
+                                final long epochDay = date.toEpochDay();
+                                if (epochDay < 0 || epochDay > Integer.MAX_VALUE) {
```

Comment:
Why don't we like negative dates?
Maybe we should support everything in `[QueryConstants.MIN_INT, Integer.MAX_VALUE]`?
Check conversions for `LocalDate.MIN` and `LocalDate.MAX`. If we can't fit this in an `int` as days, but can fit it into a `long`as millis, we should choose the other representation maybe.